src/PartialLoopPurityPass.cpp

```diff
@@ -58,22 +65,19 @@
 #elif defined(HAVE_LLVM_IR_CALLSITE_H)
 #include <llvm/IR/CallSite.h>
 #endif
-#include <llvm/Transforms/Utils/BasicBlockUtils.h>
-#include <llvm/Transforms/Utils/Cloning.h>
-#include <llvm/Analysis/ValueTracking.h>
-#include <llvm/Analysis/CallGraph.h>
-#include <llvm/IR/Verifier.h>
-#include <llvm/IR/Constants.h>
-#include <llvm/IR/InlineAsm.h>
+#include <llvm/Pass.h>
```

Comment:
The sorting w.r.t the ifdefs seems a bit arbitrary. Maybe we should sort them too based on f.ex. the first header included...

Reply:
Yes, the sorting there is arbitrary (because it is unclear how to best sort these - and we can leave this for another PR).
I decided to only apply changes until `cpplint` was happy with them.